### DISCUSSION TITLE
Where is the GModelFactory in the workflow example?

### DISCUSSION THREAD

**Discussion by gee4vee** [eclipse-glsp/glsp#598] [Q&A] [answered]
The [model generation docs](https://www.eclipse.org/glsp/documentation/modelgeneration/) mention the need for a GModelFactory implementation unique to every project. However I'm not seeing one in the Node or Java server projects. How is it working for those? Are there any examples out there of a GModelFactory implementation?

**Comment by tortmayr** [comment, marked as answer]
The Workflow example uses the so-called NullImpl of the `GModelFactory`.  There is also a section in the doc exaplaining this:

>The only exception are GLSP editors that directly operate on GModels; that is, the GModel is persisted and loaded directly by the model source loader. In such cases, no transformation from the model source and the GModel needs to be provided as the model source already is the GModel. Thus, the so-called NullImpl of the GModelFactory can be used. An example for such a use case is provided in the GLSP Workflow example.

We are working on more sophisticated examples at the moment, but as of right now there are no additional official examples available.
Your could have a look at the `GModelFactory` implementation of the coffee editor (https://github.com/eclipse-emfcloud/coffee-editor/blob/master/backend/plugins/org.eclipse.emfcloud.coffee.workflow.glsp.server/src/org/eclipse/emfcloud/coffee/workflow/glsp/server/gmodel/GModelFactory.java). However, the coffee editor uses the Java GLSP Server and EMF in combination with a ModelServer so I don't know how usefull this example is for your usecase.
